### PR TITLE
fix crashes if Ctrl+C is hit in the 'wrong' moment [v2]

### PR DESCRIPTION
**Pull request by apahim**
v1: #1165 
 - Add try/except to the `run_test()` so we can avoid crash on ctrl+c outside the test loop.

Reference: https://trello.com/c/lDj56sKW

v2:
 - Move the try/except to the `run_suite()`
 - Add `INTERRUPTED` to the summary on 'except'.

### CODE REVIEW

File: avocado/core/runner.py [hunk 1]
```diff
@@ -428,38 +428,43 @@ def run_suite(self, test_suite, mux, timeout=0, replay_map=None,
 
         index = -1
         for test_template in test_suite:
```
Comment (ldoktor) [inline avocado/core/runner.py]: Well I have just one question, what happens, when the user hits `ctrl+c` when the `for` loop assigns the new `test_template`? :wink:

Comment (apahim) [inline avocado/core/runner.py]: yep :)
Trying to understand the travis issue before send a new version with the `try` outside the `for`.

File: avocado/core/runner.py [hunk 2]
```diff
@@ -427,39 +427,43 @@ def run_suite(self, test_suite, mux, timeout=0, replay_map=None,
...
+                            test_factory = (replay_map[index], test_parameters)
+
+                        break_loop = not self.run_test(test_factory, queue,
+                                                       summary, deadline)
+                        if break_loop:
+                            break
+                runtime.CURRENT_TEST = None
+                if break_loop:
+                    break
+        except KeyboardInterrupt:
```
Comment (ldoktor) [inline avocado/core/runner.py]: Would you please also add logging.getLogger("avocado.test").error(...) to log the information about this in the result directory?

(while on it you could actually modify the whole `runner` and use the `getLogger` only once at the beginning, but we can also polish it later)

**Comment by ldoktor** [comment]
Thanks, now it seems to be working properly. I mean the inconsistency between logs and results as when the test finishes, it reports the actual result in logs, but when it's interrupted befor `end_test` the results say it was skipped.

I don't see a problem in it, because the test in deed finished, but the overall result is job INTERRUPTED so the last test's results does not matter. The only thing missing here is to log the information about interruption, because `job.log` does not contain such information, which is really confusing.

Btw I have no idea what the travis failure is, on first glance it does not look related. But I haven't looked at it properly, just glanced why is it failing when you're not changing anything major....

**Pull request closed without merging**